pkg/servicescm/servicescm.go

```diff
@@ -282,12 +282,9 @@ func (cmData *Data) ValidateExpectedContent(expected *Data) error {
 	if len(cmData.EnvironmentVars) != len(expected.EnvironmentVars) {
 		return fmt.Errorf("unexpected number of environment variable")
 	}
-	for _, expectedEnvVar := range expected.EnvironmentVars {
-		if cmData.EnvironmentVars[expectedEnvVar] != expected.EnvironmentVars[expectedEnvVar] {
-			return fmt.Errorf("required environment variable %s is not present as expected."+
-				"expected: %s, actual: %s", expectedEnvVar, expected.EnvironmentVars[expectedEnvVar],
-				cmData.EnvironmentVars[expectedEnvVar])
-		}
+	if !reflect.DeepEqual(cmData.EnvironmentVars, expected.EnvironmentVars) {
```

Comment:
Consider adding a unit test to cover the scenario where the loop failed to match the expected vars.

Reply:
thanks @jrvaldes I was working on it, updated now.